pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -279,12 +279,18 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 	if tr.Status.PodName != "" {
 		pod, err = c.KubeClientSet.CoreV1().Pods(tr.Namespace).Get(tr.Status.PodName, metav1.GetOptions{})
 		if err != nil {
-			c.Logger.Errorf("Error getting pod %q: %v", tr.Status.PodName, err)
-			return err
+			if errors.IsNotFound(err) {
+				// Reset the start time + timeout
+				tr.Status.StartTime = nil
+				c.timeoutHandler.Release(tr)
```

Comment:
im not sure we'd want to reset the timeout in this case - i think the timeout should cover execution from start to finish - however many pods had to be recreated in the meantime :thinking: